CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift

```diff
@@ -27,34 +52,57 @@ public final class DocumentService: DocumentServiceProtocol {
             .map { $0.value }
             .sorted()
             .map { $0.info }
-        
-        self.fetchExtractions(for: partialDocumentsInfoSorted, completion: completion)
+        self.analysisCancellationToken = CancellationToken()
+        captureNetworkService.analyse(partialDocuments: partialDocumentsInfoSorted, metadata: metadata, cancellationToken: analysisCancellationToken!) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .success((createdDocument, extractionResult)):
+                self.document = createdDocument
+                completion(.success(extractionResult))
+            case let .failure(error):
+                if error != .requestCancelled {
+                    completion(.failure(error))
+                }
+            }
+        }
     }
     
     public func cancelAnalysis() {
         if let compositeDocument = document {
-            delete(compositeDocument)
+            captureNetworkService.delete(document: compositeDocument) {[weak self] result in
+                switch result {
+                case .success:
+                    self?.analysisCancellationToken?.cancel()
+                    self?.analysisCancellationToken = nil
+                    self?.document = nil
+                case .failure(_):
+                    break
+                }
+            }
         }
-        
-        analysisCancellationToken?.cancel()
-        analysisCancellationToken = nil
-        document = nil
     }
     
     public func remove(document: GiniCaptureDocument) {
         if let index = partialDocuments.index(forKey: document.id) {
             if let document = partialDocuments[document.id]?
                 .document {
-                delete(document)
+                captureNetworkService.delete(document: document) {[weak self] result in
+                    switch result {
+                    case .success:
+                        self?.partialDocuments.remove(at: index)
```

Comment:
Here, too, calling the `partialDocuments.remove(at: index)` did not depend on the deletion result or whether there was a partial document for the `document.id`.

Reply:
Ok, it makes sense.